packages/components/src/utils/json.ts

```diff
@@ -1,12 +1,20 @@
 export const parseJSONAttribute = <T>(attribute: T | string): T => {
-  return typeof attribute === 'string'
-    ? // input is potentially JSON parsable string, e.g. "{ 'aria-label': 'Some label' }"
-      JSON.parse(
-        attribute
-          .replace(/(?<!\\)'/g, '"') // convert single quotes to double quotes except the ones which are escaped by backslash
-          .replace(/\\(?!u0027)/g, '') // remove string escapes except the ones followed by unicode u0027
-          .replace(/[\s"]?([\w-]+)[\s"]?:/g, '"$1":') // wrap keys in double quotes
-      )
-    : // input is object, e.g. { 'aria-label': 'Some label' }
-      attribute;
+  // Input is object, e.g. { 'aria-label': 'Some label' }
+  if (typeof attribute !== 'string') {
+    return attribute;
+  }
+
+  // Convert single quotes to double quotes except the ones which are escaped by backslash
+  let jsonString = attribute
+    .replace(/\\'/g, '__escaped_single_quote__')
+    .replace(/'/g, '"')
+    .replace(/__escaped_single_quote__/g, '\\\'');
+
+  // Remove string escapes except the ones followed by unicode u0027
+  jsonString = jsonString.replace(/([^\\])\\(?!u0027)/g, '$1');
```

Comment:
why can't we add the `replace` functions to where the others are and get rid of the let variable?

Reply:
Readability, but sure we can omit the variable

Reply:
Result (harder to read in my opinion)
```ts
return JSON.parse(
    attribute
      // Convert single quotes to double quotes except the ones which are escaped by backslash
      .replace(/\\'/g, '__escaped_single_quote__')
      .replace(/'/g, '"')
      .replace(/__escaped_single_quote__/g, '\\\'')
      // Remove string escapes except the ones followed by unicode u0027
      .replace(/([^\\])\\(?!u0027)/g, '$1')
      // Wrap keys in double quotes
      .replace(/[\s"]?([\w-]+)[\s"]?:/g, '"$1":')
  );
```

Reply:
I don't think it's more readable in this case to be honest. And I think working with `const` is always to be preferred over `let`.

Reply:
Okay, sure. I will push the proposed version above.